colour/algebra/interpolation.py

```diff
@@ -1292,7 +1292,7 @@ class PchipInterpolator(scipy.interpolate.PchipInterpolator):
     """
 
     def __init__(self, x: ArrayLike, y: ArrayLike, *args: Any, **kwargs: Any) -> None:
-        super().__init__(x, y, *args, **kwargs)
+        super().__init__(as_float_array(x), as_float_array(y), *args, **kwargs)
```

Comment:
`typing.cast` would also be an option, but strictly speaking I'd have to cast it to `Any` for it to be theoretically allowed. Downcasting this correctly would require recreating the `ToFloatND` or `ToFloat1D` types from `optype.numpy` ([docs](https://github.com/jorenham/optype#array-likes)) here, which would be rather messy.